Add feature tables

artemkovalyov opened the pull request:
## What Has Changed?

Add FeatureTable component for our feature matrix

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on it's own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

artemkovalyov commented:
To everyone concerned, I coded a component to create a feature matrix. It takes JSON as input for `features` and `columns` (table layout) and renders a table to represent our feature matrix. It's easily adaptable to any section of the docs. See examples on the overview pages of Java and JS, they respectively show only a relevant part. This is a full feature table from the overview section: https://sap.github.io/cloud-sdk/docs/overview/cloud-sdk-feature-matrix

It's fully reusable for OData and OpenAPI features as well as any others. We can keep it all together, but for less overhead, I'd suggest for bigger features create a separate JSON file with relevant fields.

The Feature Matrix is by no means complete! It's our mutual work to add items there and correct things as we go, provide a reference, notes, add documentation links when it's there, and so on. It has to be incrementally updated every time we release something new deserving to be there.

For any questions come my way and let's start the big INVENTORY!

artemkovalyov commented:
The component can be extended with Folding, Filtering, and many other neat things, but for now, it's already a breakthrough for visibility and simplicity.

artemkovalyov commented:
We'll be able to simplify this component even further after MDX will support recursive rendering of markdown within React components. Should be released soon with MDXX 2.0.

sacnl commented:
Cool.

I don't think the node-rfc lib provides any support for SOAP, does it? (regarding the note in the first row)
"OData v4 type-safe client" appears twice
What do you have in mind for CLI / Java, which shows a ✔?

artemkovalyov commented:
> Cool.
Thanks @sacnl 
> 
> I don't think the node-rfc lib provides any support for SOAP, does it? (regarding the note in the first row)
I didn't thoroughly checked. I think you're right.

> "OData v4 type-safe client" appears twice
Fixed.

> What do you have in mind for CLI / Java, which shows a ?
Added a note about the Maven plugin.

We have plenty of scattered featured that have no coherent naming, thorough scoping, etc. If we don't know or understand some of them how customers would? My goal was to make the first draft so that our subject-matter experts take time to review and extend it. My role is to coordinate, enable, and moderate it so that we're all on the same page.

It'll require some backlog if not happening ad-hoc as a part of DoDs, etc.

FrankEssenberger commented on docs/components/JavaFeatureTable.js:
```diff
@@ -0,0 +1,69 @@
+import gfm from 'remark-gfm';
+import emoji from 'remark-emoji';
+
+function Table({ columns, data }) {
```
This function looks much like the one in the JSFeatureTable perhaps you can reuse it?

FrankEssenberger commented on docs/components/JsFeatureTable.js:
```diff
@@ -0,0 +1,69 @@
+  );
+}
+
+export default function JsFeatureTable() {
```
Add a little doc to this and the java version that the content of the table is read from the features file which hava a `js` and `java` section. You are of course able see this from the code but a little explaination always helps

FrankEssenberger commented on docs/components/data/features.js:
```diff
@@ -13,8 +13,7 @@ export const features = [
     js: {
       status: ':x:',
       docsLink: ``,
-      note:
```
Could you add a checkbox in the PR template that you adjusted the features file if you added documentation on a new feature.

FrankEssenberger reviewed (changes requested):
looks good only some minor suggestions.

FrankEssenberger commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -5,10 +5,11 @@ Explain what you have changed.
 - [ ] You formatted all changed files with prettier (`npm run prettier`)
 - [ ] You tested if the documentation still builds (`npm run build`)
 - [ ] You verified all new and changed links still work (changing the `id` or name of a file can break links)
+- [ ] You have updated a [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.
```
```suggestion
- [ ] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.
```

FrankEssenberger reviewed (approved):
LGTM
